src/CanvasRenderingContext2d.cc

```diff
@@ -662,10 +663,13 @@ NAN_METHOD(Context2d::PutImageData) {
         uint8_t b = *srcRow++;
         uint8_t a = *srcRow++;
 
-        // argb
-        // performance optimization: fully transparent/opaque pixels can be
-        // processed more efficiently.
-        if (a == 0) {
+        if (dstRow < dstStart) {
```

Comment:
No problem, I'm happy to work on a fix for this. Can probably do it this weekend. Thanks again for finding the issue.

Re: publishing, I don't have the ability to make releases, I just help out with issues where I can, and I'm not sure if there are more breaking changes that should go in before 2.0 is officially released.